frontend/src/pages/profile/tokens.js

```diff
@@ -86,134 +47,102 @@ export class UserTokens extends React.Component {
-    const pagination = {
-      pageSize: this.state.pageSize,
-      page: this.state.page,
-      totalItems: this.state.totalItems
-    };
-    return (
-      <React.Fragment>
-        <PageSection id="page" variant={PageSectionVariants.light}>
-          <Flex justifyContent={{ default: 'justifyContentSpaceBetween' }}>
-            <FlexItem spacer={{ default: 'spacerLg' }}>
-              <TextContent>
-                <Text className="title" component="h1">Tokens</Text>
-              </TextContent>
-            </FlexItem>
-            <FlexItem>
-              <Button
-                aria-label="Add token"
-                variant="secondary"
-                title="Add token"
-                onClick={this.onAddTokenClick}
-              >
-                <PlusCircleIcon /> Add Token
-              </Button>
-            </FlexItem>
-          </Flex>
-        </PageSection>
-        <PageSection>
-          <Card>
-            <CardBody className="pf-u-p-0">
-              <FilterTable
-                columns={columns}
-                rows={rows}
-                pagination={pagination}
-                isEmpty={this.state.isEmpty}
-                isError={this.state.isError}
-                onSetPage={this.setPage}
-                onSetPageSize={this.setPageSize}
-              />
-            </CardBody>
-          </Card>
-        </PageSection>
-        <AddTokenModal
-          isOpen={this.state.isAddTokenOpen}
-          onClose={this.onAddTokenClose}
-        />
-        <DeleteModal
-          title="Delete token"
-          body="Would you like to delete the selected token?"
-          toDeleteId={this.state.tokenToDelete?.id}
-          toDeletePath={['user', 'token']}
-          isOpen={this.state.isDeleteTokenOpen}
-          onClose={this.onDeleteTokenClose}
-        />
-        <ToastContainer autoClose={ALERT_TIMEOUT} stacked/>
-      </React.Fragment>
-    );
-  }
-}
+  }, [page, pageSize, isAddTokenOpen, isDeleteTokenOpen]); // extra deps to trigger fetch on modal close
+
+
+  document.title = 'User Tokens | Ibutsu';
+  return (
+    <React.Fragment>
+      <PageSection id="page" variant={PageSectionVariants.light}>
+        <Flex justifyContent={{ default: 'justifyContentSpaceBetween' }}>
+          <FlexItem spacer={{ default: 'spacerLg' }}>
+            <TextContent>
+              <Title headingLevel="h1">Tokens</Title>
+            </TextContent>
+          </FlexItem>
+          <FlexItem>
+            <Button
+              aria-label="Add token"
+              variant="secondary"
+              title="Add token"
+              onClick={() => setIsAddTokenOpen(true)}
+            >
+              <PlusCircleIcon /> Add Token
+            </Button>
+          </FlexItem>
+        </Flex>
+      </PageSection>
+      <PageSection>
+        <Card>
+          <CardBody className="pf-u-p-0">
+            <FilterTable
+              columns={COLUMNS}
+              rows={tokens.map((t) => tokenToRow(t))}
```

Comment:
@mshriver it will not work with initial state defined as [getSpinnerRow(4)] as it will try to apply tokenToRow helper functon on it. Currently if you load the page - you will not see spinner. it might be not visible as it happens fast, but to test it you can put `debugger;` before HTTP request here `HttpClient.get([Settings.serverUrl, 'user', 'token'], {page: page, pageSize: pageSize})`